src/commands/agent/generate/template.ts

```diff
@@ -31,32 +62,132 @@ export default class AgentGenerateTemplate extends SfCommand<AgentGenerateTempla
+    // TODO: Should we add a flag to specify the packagePath?
+    const packagePath = this.project?.getDefaultPackage().fullPath;
+    const basePath = join(packagePath as string, 'main', 'default');
+    const botDir = join(basePath, 'bots', botName);
+    const genAiPlannerDir = join(basePath, 'genAiPlanners');
+    const botTemplateDir = join(basePath, 'botTemplates');
 
-    writeFileSync(agentTemplateFilePath, xmlContent);
+    // Parse the metadata files as JSON
+    const botJson = xmlToJson<BotExt>(join(botDir, `${botName}.bot-meta.xml`), parser);
+    const botVersionJson = xmlToJson<BotVersionExt>(join(botDir, `v${botVersion}.botVersion-meta.xml`), parser);
+    const genAiPlannerMetaJson = xmlToJson<GenAiPlannerExt>(
+      join(genAiPlannerDir, `${botName}.genAiPlanner-meta.xml`),
+      parser
+    );
 
-    this.log(`\nSaved agent template: ${agentTemplateFilePath}`);
+    // Modify the metadata files for final output
+    // TODO: Confirm this name (might be conversationDefinitionPlanners)
+    genAiPlannerMetaJson.GenAiPlanner.botTemplate = finalFilename;
+    const botTemplate = convertBotToBotTemplate(botJson, botVersionJson, finalFilename);
 
-    return { path: agentTemplateFilePath };
+    // Build and save the metadata files
+    const cloneGenAiPlannerFilePath = join(genAiPlannerDir, `${finalFilename}.genAiPlanner-meta.xml`);
+    jsonToXml<GenAiPlannerExt>(cloneGenAiPlannerFilePath, genAiPlannerMetaJson, builder);
+
+    const botTemplateFilePath = join(botTemplateDir, `${finalFilename}.botTemplate-meta.xml`);
+    jsonToXml<BotTemplateExt>(botTemplateFilePath, botTemplate, builder);
+
+    this.log(`\nSaved BotTemplate to:\n - ${botTemplateFilePath}`);
+    this.log(`Saved GenAiPlanner to:\n - ${cloneGenAiPlannerFilePath}`);
+
+    return {
+      genAiPlannerPath: cloneGenAiPlannerFilePath,
+      botTemplatePath: botTemplateFilePath,
+    };
   }
 }
 
-const xmlContent = `<?xml version="1.0" encoding="UTF-8"?>
-<AgentTemplate xmlns="http://soap.sforce.com/2006/04/metadata">
-</AgentTemplate>
-`;
+const convertBotToBotTemplate = (
+  bot: BotExt,
+  botVersionJson: BotVersionExt,
+  newMlDomainName: string
+): BotTemplateExt => {
+  const entryDialog = botVersionJson.BotVersion.entryDialog;
+  const { conversationSystemDialogs } = botVersionJson.BotVersion;
+
+  // We need to pull the botDialog from the BotVersion file that matches the entryDialog
+  // This will be added to the BotTemplate
+  const entryDialogJson = botVersionJson.BotVersion.botDialogs.find((dialog) => dialog.developerName === entryDialog);
+
+  if (!entryDialogJson) throw new SfError('No entryDialog found in BotVersion file');
+  // TODO: Test this on a newer org. I had to have this renamed.
+  entryDialogJson.label = entryDialog;
+
+  if (!bot.Bot.label) throw new SfError('No label found in Agent (Bot) file');
+  if (!bot.Bot.botMlDomain) throw new SfError('No botMlDomain found in Agent (Bot)');
```

Comment:
A nice enhancement to these error messages would be the file path to the bot.

Reply:
I did have the file listed but removed it to reduce the args. I can add it back